accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,395 @@
+- whether Management Node(s) can be managed by Uyuni/SUSE Manager (specifically: can users manage the updates of the Management Node via Uyuni/SUSE Manager?)
+
+#### Cluster bootstrapping
+
+The cluster must be initiated by the Management Node that has all the tools to bootstrap and manage the cluster. Specific instructions on how to bootstrap the cluster must be provided.
+Questions that need to be answered by the cluster provider:
+- Upon successful bootstrapping, can the worker nodes be managed by Uyuni/SUSE Manager?
+- Do the nodes have a specific agent running on them and/or can we install a `salt-minion` or use `salt-ssh` on those nodes?
+- Can the nodes be managed by Uyuni/SUSE Manager? Can the packages be updated or do they require specific handling by the Management Node?
+
+### Manage a cluster using Uyuni/SUSE Manager
+
+Uyuni/SUSE Manager offers to show the status of the running cluster by either accessing the cluster directly or via the Management Node.
+In this regard, the cluster provider should provide a way to expose the cluster status (e.g. running nodes, the health of each node, monitoring metrics...) to Uyuni/SUSE Manager.
+
+#### Add and removal of nodes to the cluster
+
+Uyuni/SUSE Manager also offers the possibility of adding and removing nodes to the cluster:
+
+- To add a node from scratch, an AutoYaST profile can be reused
+- To remove a node, Uyuni/SUSE Manager must remove the node from the cluster using the Management Node and issuing the proper procedure
+
+Uyuni/SUSE Manager should be instructed to perform additional actions depending on the role of added/removed node (e.g. control plane, worker, load balancer, ...).
+
+#### Cluster upgrade
+
+If every node of the cluster is not directly registered to Uyuni/SUSE Manager, Uyuni/SUSE Manager should offer the possibility to use the Management Node to trigger an update of every node in the cluster, following the standard procedure of upgrading predisposed by the cluster provider.
+
+Also, Uyuni/SUSE Manager must coordinate the Management Node to issue a correct upgrade procedure of the product (e.g. from CaaS Platform version 3 to 4).
+
+#### Application deployment
+
+Uyuni/SUSE Manager can also offer the possibility of deploying an application on top of the cluster deployed (e.g. Kubernetes deployment descriptor to deploy a Kubernetes operator). In this case, the cluster provider must provide the Management Node of the necessary tooling to deploy an application using an API. Uyuni/SUSE Manager makes use of the Management Node to deploy a provided application.
+
+#### Specific usage of Uyuni/SUSE Manager features
+
+One cluster deployment can also leverage Uyuni/SUSE Manager features via API.
+For example, a cluster deployment may invoke `CVEAuditHandler` and test if a container image built by SUSE Manager is vulnerable against a particular CVE and decide whether to allow the deployment of the image or not.
+
```

Comment:
Consider the example: CaaSP uses SUSE Manager to build a container image. SUSE Manager can check if the image is vulnerable against a CVE.
Using this information, CaaSP can decide whether to provision (or not) an application that uses a vulnerable image by invoking SUSE Manager's API.

SOC, for example, can proactively check if one of the running nodes is vulnerable to a certain CVE audit and take countermeasures (swap node?).

Reply:
In your caasp example, how will you build a container image, and change the manifests of the caasp deployment so that this image is taken instead of the default one? How do you make sure that this is supported (assuming ppl can inject ANYTHING in the image)? Then, assuming the customer overrides the image for v1.2.0, how do you make sure the v1.2.1 or v1.3.0 will also be created by the customer, to allow updates/upgrades?

The rest makes sense, it's possible for a kubernetes based product to refuse to deploy based on a webhook call.

Reply:
In the example, CaaSP user must change the manifest of the image deployed (if the image tag is changed). If the tag remains the same (because it is just a rebuild) then there is no need to change the manifest.
To make sure that the "new" image is used, Uyuni/SUSE Manager and CaaSP must use the same container registry, ideally with authentication.
To make sure that the images are created by the customer, I am assuming that the images are all built using Uyuni/SUSE Manager using the image building feature.

Reply:
How would that work with _supporting_ OpenStack? In our example, we have deep links between libraries versions. Two cases:
1) Assuming someone overrides the image with the same tag, deployment will indeed be easy. However, support will be hard: How can we tell if the newly built image is supported or not? Should mark all the custom built images as not supported? We already have to deal with a very large amount of supported features...
2) Assuming someone overrides the image with a new tag, deployment will have to be configured. Do you expect the customer to have a deep understanding of the k8s manifests for openstack, and how they integrate with each other? In that case, does that sound right that the customer has to edit tens of manifests to update said image? Do you propose a solution for this? How do you ensure that the tag is not overlapping with our next version tag? All these questions have to be answered for me to understand if the image building feature is a wise choice for SOC. I assume that, for CaaSP, this doesn't have to be answered. But please allow other teams to disable that feature.

Reply:
I wouldn't disable the feature per se, but rather in the product (in this case SOC) there is no need to invoke the CVE Audit API call before deployments, for the reasons you clearly stated.
In the RFC we say "the deployment can also leverage Uyuni/SUSE Manager via API" but it is not required to do so.
Do you think we should rephrase it?